File: AtCoder/AtCoder_Beginner_Contest_154/D_Dice_in_Line.cpp
#include <stdio.h>
using namespace std;

int main(){
    int N, K;
    int *p_list;
    double max_val = 0, E_sum;

    scanf("%d %d", &N, &K);
    p_list = new int[N];

    for (int i = 0; i < N; i++){
        scanf("%d", &p_list[i]);
    }

    for (int i = 0; i < N - K + 1; i++){
        E_sum = 0;
        for (int j = 0; j < K; j++){
            E_sum += p_list[i + j];
        }
        E_sum = (E_sum + K) / 2.0;
        if (max_val < E_sum){
            max_val = E_sum;
        }
    }
    printf("%lf\n", max_val);

    delete[] p_list;
    return 0;
}

File: AtCoder/AtCoder_Beginner_Contest_154/D_Dice_in_Line.py
N, K = list(map(int, input().split(' ')))
p_list = list(map(int, input().split(' ')))
max_val = 0
for i in range(N - K + 1):
    E_sum = 0
    for j in range(K):
        index = i + j
        E_sum += p_list[index]
    E_sum = (E_sum + K) / 2.0
    if max_val < E_sum:
        max_val = E_sum
print(max_val)